File: print_date/print_date.py
from print_date.calendar import Calendar
from print_date.printer import Printer


class PrintDate:
    
    def __init__(self, calendar: Calendar, printer: Printer):
        self._calendar = calendar
        self._printer = printer
    
    def print_current_date(self) -> None:
        line = str(self._calendar.today())
        self._printer.print_line(line)

File: src/print_date/PrintDate.ts
import { Calendar } from './Calendar';
import { Printer } from './Printer';

export class PrintDate {
  private readonly calendar: Calendar;
  private readonly printer: Printer;

  constructor(calendar: Calendar, printer: Printer) {
    this.calendar = calendar;
    this.printer = printer;
  }

  printCurrentDate(): void {
    const line = this.calendar.today().toString();
    this.printer.printLine(line);
  }
}
